### PR TITLE
Add GraphQL metrics handing.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Add handling for playbook metrics in GraphQL endpoint.

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Where can we look at the checksums to verify?

File: server/sqlstore/playbook.go [hunk 1]
```diff
@@ -846,6 +862,84 @@ func (p *playbookStore) GetAutoFollows(playbookID string) ([]string, error) {
...
+		return errors.New("id should not be empty")
+	}
+
+	_, err := p.store.execBuilder(p.store.db, sq.
+		Update("IR_MetricConfig").
+		Set("DeleteAt", model.GetMillis()).
+		Where(sq.Eq{"ID": id}))
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete metric with id '%s'", id)
```
Comment (agarciamontoro) [inline server/sqlstore/playbook.go:920]: Nit:
```suggestion
		return errors.Wrapf(err, "failed to delete metric with id %q", id)
```

File: server/sqlstore/playbook.go [hunk 2]
```diff
@@ -846,6 +862,84 @@ func (p *playbookStore) GetAutoFollows(playbookID string) ([]string, error) {
...
+		return errors.New("id should not be empty")
+	}
+
+	_, err := p.store.execBuilder(p.store.db, sq.
+		Update("IR_MetricConfig").
+		SetMap(setmap).
+		Where(sq.Eq{"ID": id}))
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to update metric with id '%s'", id)
```
Comment (agarciamontoro) [inline server/sqlstore/playbook.go:937]: Same nit as before
```suggestion
		return errors.Wrapf(err, "failed to update metric with id %q", id)
```

File: server/sqlstore/playbook.go [hunk 3]
```diff
@@ -846,6 +862,84 @@ func (p *playbookStore) GetAutoFollows(playbookID string) ([]string, error) {
...
+}
+
+func (p *playbookStore) AddMetric(playbookID string, config app.PlaybookMetricConfig) error {
+	numExistingMetrics, err := p.GetNumMetrics(playbookID)
+	if err != nil {
+		return err
+	}
+
+	if numExistingMetrics >= app.MaxMetricsPerPlaybook {
+		return errors.Errorf(fmt.Sprintf("playbook cannot have more than %d key metrics", app.MaxMetricsPerPlaybook))
```
Comment (agarciamontoro) [inline server/sqlstore/playbook.go:894]: There's no need for `fmt.Sprintf` here
```suggestion
		return errors.Errorf("playbook cannot have more than %d key metrics", app.MaxMetricsPerPlaybook)
```

File: server/api_graphql_playbooks_test.go [hunk 4]
```diff
@@ -141,3 +104,122 @@ mutation UpdatePlaybook($id: String!, $updates: PlaybookUpdates!) {
...
+			OperationName: "AddMetric",
+			Variables: map[string]interface{}{
+				"playbookID":  e.BasicPlaybook.ID,
+				"title":       "New Metric",
+				"description": "the description",
+				"type":        app.MetricTypeDuration,
+			},
+		}, &response)
+		require.NoError(t, err)
+		require.Len(t, response.Errors, 0)
```
Comment (agarciamontoro) [inline server/api_graphql_playbooks_test.go:168]: Super nit:
```suggestion
		require.Empty(t, response.Errors)
```

File: server/api_graphql_playbooks_test.go [hunk 5]
```diff
@@ -141,3 +104,122 @@ mutation UpdatePlaybook($id: String!, $updates: PlaybookUpdates!) {
...
+		err := e.PlaybooksClient.DoGraphql(context.Background(), &client.GraphQLInput{
+			Query:         testUpdateMetricQuery,
+			OperationName: "UpdateMetric",
+			Variables: map[string]interface{}{
+				"id":    e.BasicPlaybook.Metrics[0].ID,
+				"title": "Updated Title",
+			},
+		}, &response)
+		require.NoError(t, err)
+		require.Len(t, response.Errors, 0)
```
Comment (agarciamontoro) [inline server/api_graphql_playbooks_test.go:194]: Super nit (x2)
```suggestion
		require.Empty(t, response.Errors)
```

File: server/api_graphql_playbooks_test.go [hunk 6]
```diff
@@ -141,3 +104,122 @@ mutation UpdatePlaybook($id: String!, $updates: PlaybookUpdates!) {
...
+		var response graphql.Response
+		err := e.PlaybooksClient.DoGraphql(context.Background(), &client.GraphQLInput{
+			Query:         testDeleteMetricQuery,
+			OperationName: "DeleteMetric",
+			Variables: map[string]interface{}{
+				"id": e.BasicPlaybook.Metrics[0].ID,
+			},
+		}, &response)
+		require.NoError(t, err)
+		require.Len(t, response.Errors, 0)
```
Comment (agarciamontoro) [inline server/api_graphql_playbooks_test.go:218]: Super nit (x3)
```suggestion
		require.Empty(t, response.Errors)
```

**Review by agarciamontoro — APPROVED** [review]
So cool to see GraphQL PRs :nerd_face: I left a bunch of nits and a doubt on `UpdateMetric`, but nothing blocking. Thank you :tada:

**Review by calebroseland — APPROVED** [review]

File: server/sqlstore/playbook.go [hunk 7]
```diff
@@ -846,6 +862,84 @@ func (p *playbookStore) GetAutoFollows(playbookID string) ([]string, error) {
...
+		Where(sq.Eq{"ID": id}))
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete metric with id '%s'", id)
+	}
+
+	return nil
+}
+
+func (p *playbookStore) UpdateMetric(id string, setmap map[string]interface{}) error {
```
[the remark this answers is not in the document]
Comment (crspeller) [reply server/sqlstore/playbook.go:926]: No validation. It's not functionally different then what we already do. 
However @lieut-data had the same concern. I think I will refactor this along with the GraphqlUpdate function in another PR to move more of the validation into the store layer.

**Pull request merged**